src/AppInstallerCommonCore/MSStore.cpp

```diff
@@ -141,16 +156,27 @@ namespace AppInstaller::MSStore
                 if (existingItemResult == CheckExistingItemResult::Cancel)
                 {
                     installManager.Cancel(productId);
+
+                    // Wait for at most 10 seconds for install item to be removed from queue.
+                    for (int i = 0; i < 20; ++i)
+                    {
+                        Sleep(500);

```

Comment:
```suggestion
                        Sleep(200);
```
And the appropriate loop count to get back to 10 seconds.